e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java

```diff
@@ -100,23 +84,55 @@ public void When_ExtractRequestReceivedForPatientWithNoDocs_Expect_DatabaseToBeU
     @Test
     public void When_ExtractRequestReceivedForNotExistingPatient_Expect_ErrorUpdatedInDatabase() throws Exception {
         String conversationId = UUID.randomUUID().toString();
-        String ehrExtractRequest = buildEhrExtractRequest(conversationId, NOT_EXISTING_PATIENT_NHS_NUMBER);
+        String ehrExtractRequest = buildEhrExtractRequest(conversationId, NOT_EXISTING_PATIENT_NHS_NUMBER, FROM_ODS_CODE);
 
         MessageQueue.sendToMhsInboundQueue(ehrExtractRequest);
 
         var ehrExtractStatus = waitFor(() -> Mongo.findEhrExtractStatus(conversationId));
-        assertThatInitialRecordWasCreated(conversationId, ehrExtractStatus, NOT_EXISTING_PATIENT_NHS_NUMBER);
+        assertThatInitialRecordWasCreated(conversationId, ehrExtractStatus, NOT_EXISTING_PATIENT_NHS_NUMBER, FROM_ODS_CODE);
 
         var ackToRequester = (Document) waitFor(() -> Mongo.findEhrExtractStatus(conversationId).get("ackToRequester"));
         assertThatNegativeAcknowledgementToRequesterWasSent(ackToRequester, NEGATIVE_ACKNOWLEDGEMENT_TYPE_CODE);
         assertThatErrorInfoIsStored(conversationId, GET_GPC_STRUCTURED_TASK_NAME);
     }
 
-    private String buildEhrExtractRequest(String conversationId, String notExistingPatientNhsNumber) throws IOException {
+    @Test
+    public void When_ExtractRequestReceivedDifferentGPCProvider_Expect_ExtractStatusAndDocumentDataAddedToDatabase() throws Exception {
+        String conversationId = UUID.randomUUID().toString();
+        String ehrExtractRequest = buildEhrExtractRequest(conversationId, EXISTING_PATIENT_NHS_NUMBER, FROM_ODS_CODE2);
+        MessageQueue.sendToMhsInboundQueue(ehrExtractRequest);
+
+        assertHappyPathWithDocs(conversationId, FROM_ODS_CODE2);
+    }
```

Comment:
This test relies on the existence of another test, which uses a different value for `from ODS code`. In order to make it independent, maybe we should send two messages in it, each with a different ODS code and a different conversation ID and then have two assertions like this:
```
assertHappyPathWithDocs(conversationId1, FROM_ODS_CODE);
assertHappyPathWithDocs(conversationId2, FROM_ODS_CODE2);
```

Reply:
Our first test in e2e sends the same thing but different ods code, however with our await when sending two in one test there is a big delay so I thought this was the best option. What do you think?

Reply:
If we send both messages before running the assertions, the combined wait time should be similar (but we can check how big the difference is).

Tests need to be independent from each other, so that a change in one test doesn't affect any other. In this case, the removal of the other test would make this one incomplete, but it wouldn't be obvious - the test would still be passing.